### PR TITLE
Adds 1.3.10 release notes

### PR DESCRIPTION
**Pull request by ryanbogan**
### Description
Adds 1.3.10 release notes

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch.release-notes-1.3.10.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## 2023-05-18 Version 1.3.10 Release Notes
+
+### Upgrades
+- Upgrade Netty from 4.1.90.Final to 4.1.91.Final , ASM 9.4 to ASM 9.5, ByteBuddy 1.14.2 to 1.14.3 ([#6981](https://github.com/opensearch-project/OpenSearch/pull/6981))
+- Upgrade `org.gradle.test-retry` from 1.5.1 to 1.5.2
+- Upgrade `org.apache.hadoop:hadoop-minicluster` from 3.3.4 to 3.3.5
+- OpenJDK Update (April 2023 Patch releases) [#7449](https://github.com/opensearch-project/OpenSearch/pull/7449)
+- Upgrade `net.minidev:json-smart` from 2.4.7 to 2.4.10
+- Security: https://github.com/opensearch-project/OpenSearch/compare/1.3.9...HEAD
```
Comment (Poojita-Raj) [inline release-notes/opensearch.release-notes-1.3.10.md:9]: Nit: Could you rephrase this line on security upgrade to provide more context?

Comment (ryanbogan) [reply release-notes/opensearch.release-notes-1.3.10.md]: Should be fixed now

**Review by Poojita-Raj — APPROVED** [review]

File: release-notes/opensearch.release-notes-1.3.10.md (hunk 1 above)
Comment (Poojita-Raj) [reply release-notes/opensearch.release-notes-1.3.10.md]: Feel free to resolve this

File: release-notes/opensearch.release-notes-1.3.10.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+## 2023-05-18 Version 1.3.10 Release Notes
+
+### Upgrades
+- Upgrade Netty from 4.1.90.Final to 4.1.91.Final , ASM 9.4 to ASM 9.5, ByteBuddy 1.14.2 to 1.14.3 ([#6981](https://github.com/opensearch-project/OpenSearch/pull/6981))
+- Upgrade `org.gradle.test-retry` from 1.5.1 to 1.5.2
+- Upgrade `org.apache.hadoop:hadoop-minicluster` from 3.3.4 to 3.3.5
```
Comment (owaiskazi19) [inline release-notes/opensearch.release-notes-1.3.10.md:6]: [Snakeyaml](https://github.com/opensearch-project/OpenSearch/blob/1.x/CHANGELOG.md?plain=1#L11) entry is missing here
